### PR TITLE
Enable pylint w0703

### PR DESCRIPTION
**Pull request by richtja**
This pylint checks if there is a catch of too general exception.

Signed-off-by: Jan Richter <jarichte@redhat.com>

### CODE REVIEW

File: avocado/core/extension_manager.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def __init__(self, namespace, invoke_kwds=None):
             try:
                 plugin = ep.load()
                 obj = plugin(**invoke_kwds)
-            except Exception as exception:
```
Comment (clebergnu) [inline avocado/core/extension_manager.py]: We need a `# pylint: disable=W0703` here instead, as the plugin instantiation can fail due to many other causes, not only an `ImportError`.

File: avocado/core/job.py [hunk 2]
```diff
@@ -240,10 +240,14 @@ def _setup_job_category(self):
         try:
             os.symlink(os.path.relpath(self.logdir, category_path),
                        os.path.join(category_path, os.path.basename(self.logdir)))
-        except Exception:
+        except NotImplementedError:
```
Comment (clebergnu) [inline avocado/core/job.py]: I don't quite understand where the `NotImplementedError` can be raised from.  The most sensitive operation here seems to be `os.symlink`, which according to https://docs.python.org/3/library/os.html#os.symlink doesn't raise it.

**Comment by richtja** [comment]
Thanks for the comments, I know that there is many variables which can be influenced by this PR. So if you find anything, let me know I will fix that.

File: avocado/core/job.py (hunk 2 above)
Comment (richtja) [reply avocado/core/job.py]: Yes, I read that and found out that on the older versions of windows it raise `NotImplementedError`. I do not really sure what is the support of avocado for such systems so I was thinking that will be better to have it there.

**Comment by clebergnu** [comment]
Merged as 2ec75c7fdc992a21457ab8e495d5158e8c11fb35

**Pull request closed without merging**